tests/hpc/slurm_master_db.pm

```diff
@@ -18,6 +18,165 @@ use warnings;
 use testapi;
 use lockapi;
 use utils;
+use Data::Dumper;
+use Mojo::JSON;
+
+## TODO: provide better parser for HPC specific tests
+sub get_test_name {
+    return my $name = (caller(2))[3];
+}
+
+sub validate_result {
+    my ($result) = @_;
+
+    if ($result == 0) {
+        return 'PASS';
+    } elsif ($result == 1) {
+        return 'FAIL';
+    } else {
+        return undef;
+    }
+}
+
+sub results {
+    my ($description, $result) = @_;
+
+    my %results = (
+        test        => get_test_name(),
+        description => $description,
+        result      => validate_result($result)
+    );
+    return %results;
+}
+
+sub pars_results {
+    my (%test)  = @_;
+    my $file    = 'tmpresults.xml';
+    my @results = keys %test;
+
+    my $test_name;
+    my $description;
+    my $result;
+
+    for my $i (@results) {
+        if ($i eq 'test') {
+            $test_name = $test{$i};
+        } elsif ($i eq 'description') {
+            $description = $test{$i};
+        } elsif ($i eq 'result') {
+            if ($test{$i} eq 'FAIL') {
+                $result = $test{$i};
+            }
+        } else {
+            die('Wrong key!');
+        }
+    }
+
+    if ($result eq 'FAIL') {
+        script_run("echo \"<testcase name='$test_name' errors='1'>\" | tee -a $file");
+    } else {
+        script_run("echo \"<testcase name='$test_name'>\" | tee -a $file");
+    }
+    script_run("echo \"<system-out>\" | tee -a $file");
+    script_run("echo $description | tee -a $file");
+    script_run("echo \"</system-out>\" | tee -a $file");
+    script_run("echo \"</testcase>\" | tee -a $file");
+}
+
+sub test_01 {
+    my $description = 'Basic SRUN test';
+    my $result;
+
+    $result = script_run("srun -w slave-node00 date");
+
+    my %results = results($description, $result);
+    return %results;
+}
+
+sub test_02 {
+    my $description = 'Simple SINFO test';
+    my $result;
+
+    $result = script_run('sinfo');
+
+    ##TODO: add check of sinfo vs slurm.conf?
+
+    my %results = results($description, $result);
+    return %results;
+}
+
+##TODO: improve before enabling
+sub test_03 {
+    my $description = 'SRUN stress test';
+    my $result      = 0;
+    my $result0;
+    my $result1;
+
+    my @collected_results;
+
+    for (my $i = 0; $i < 50; $i++) {
+        $result0 = script_run("srun -w slave-node00 date");
+        push(@collected_results, $result0);
+        $result1 = script_run("srun -w slave-node01 date");
+        push(@collected_results, $result1);
+    }
+
+    foreach my $i (@collected_results) {
+        if ($i == 1) {
```

Comment:
```suggestion
       $result = 1 if $i == 1;
```

Reply:
I should perhaps not push this. This test is disabled anyway and I was adding this: ##TODO: improve before enabling
In essence: scheduled jobs can fail... occasionally. This happens and is OK from slurm stand-point as far as I understand it (even the code handles such cases).

The point here would be to see the ration of such drops and check for the variance over the runs... but I am not clear if I don't over-engineer here. 

I will replace the code, which I spend on a few minutes only, and replace with TODO: implement stress test.